assets/src/components/AssignmentTable.js

```diff
@@ -120,170 +144,272 @@ function AssignmentTable (props) {
     }
   }, [currentWeekRow.current])
 
+  const [assignmentGroupNames, setAssignmentGroupNames] = useState([])
+  useEffect(() => {
+    const allGroupNames = assignments.map(ag => ag.assignmentGroup.name)
+    var uniqueGroupNames = [...new Set(allGroupNames)]
+    uniqueGroupNames.sort()
+    setAssignmentGroupNames(uniqueGroupNames)
+  }, [assignments])
+
+  const [assignmentFilter, setAssignmentFilter] = useState('')
+  function handleNameFilterChange (e) {
+    setAssignmentFilter(e.target.value)
+  }
+
+  const ITEM_HEIGHT = 48
+  const ITEM_PADDING_TOP = 8
+  const MenuProps = {
+    PaperProps: {
+      style: {
+        maxHeight: ITEM_HEIGHT * 4.5 + ITEM_PADDING_TOP,
+        width: 250
+      }
+    }
```

Comment:
I am asking this for my knowledge ... I see that this is used in the `Select` element, and `MenuProps` is an object and how exactly these are applied to the element

Reply:
It is how properties are set on the Select element in the  [material UI multi select](https://material-ui.com/components/selects/#multiple-select) documentation.